src/Microsoft.Health.Fhir.Shared.Api/Features/Resources/Bundle/BundleHandler.cs

```diff
@@ -280,7 +279,28 @@ public async Task ResolveBundleReferences(EntryComponent entry, Dictionary<strin
                 // Checks to see if this reference has already been assigned an Id
                 if (referenceIdDictionary.TryGetValue(reference.Reference, out var referenceInformation))
                 {
-                    reference.Reference = $"{referenceInformation.resourceType}/{referenceInformation.resourceId}";
+                    if (!reference.Reference.Contains("/_history/", StringComparison.OrdinalIgnoreCase))
+                    {
+                        reference.Reference = $"{referenceInformation.resourceType}/{referenceInformation.resourceId}";
+                    }
+                    else
+                    {
+                        var version = await _transactionBundleValidator.GetLatestVersionId(new ResourceKey(referenceInformation.resourceType, referenceInformation.resourceId), cancellationToken);
+
+                        string updatedVersionId = "1";
+
+                        // updates version after this transaction is complete.
+                        // if this is a new resource, version will be null during transaction and therefore will default to 1 after transaction.
+                        if (version != null)
+                        {
+                            if (int.TryParse(version, out int versionId))
+                            {
+                                updatedVersionId = (versionId + 1).ToString();
+                            }
+                        }
+
+                        reference.Reference = $"{referenceInformation.resourceType}/{referenceInformation.resourceId}/_history/{updatedVersionId}";
```

Comment:
Is there a case here where the same reference exists in the same bundle more than once but could get a different version because the referenced resource was updated out of band?

Reply:
Can you clarify more on this?

Reply:
Implementation has been changed. The new checkin doesnot have this code flow.